### DISCUSSION TITLE
it's mandatory to install visual studio?

### DISCUSSION THREAD

**Discussion by coxato** [nexe/nexe#895] [Q&A]
Hi, I have a problem when I run `nexe --build index.js`, so I add `--verbose` and this is the message:

![terminal](https://user-images.githubusercontent.com/34953669/117556703-ba4a7b80-b039-11eb-99a1-f67310739d2e.jpg)

the message said: **failed to find a suitable Visual studio installation** and also **vcbuild.bat nosign release x64 exited with code: 1**

### My question is:
is it mandatory to have visual studio installed? I read that "desktop development with c++" is necessary to make the .exe in windows. BUT:

![visual studio](https://user-images.githubusercontent.com/34953669/117556884-a43dba80-b03b-11eb-8f5f-e4dd434a0809.jpg)

 I make this question because my internet connection is so bad, and the visual studio installation will take me 3 or 4 days, because it's a little bit more of 7gb

**Comment by Aworldc** [comment]
i'm not sure if i understand your question, but if you just want to compile a js file, and don't actually need to compile nexe with it, you can just omit the `--build` flag and run `nexe index.js`
